restapi/client.go

```diff
@@ -133,13 +134,45 @@ func (c mcsCredentials) Expire() {
 	c.minioCredentials.Expire()
 }
 
+// mcsSTSAssumeRole it's a STSAssumeRole wrapper, in general
+// there's no need to use this struct anywhere else in the project, it's only required
+// for passing a custom *http.Client to *credentials.STSAssumeRole
+type mcsSTSAssumeRole struct {
+	stsAssumeRole *credentials.STSAssumeRole
+}
+
+func (s mcsSTSAssumeRole) Retrieve() (credentials.Value, error) {
+	return s.stsAssumeRole.Retrieve()
+}
+
+func (s mcsSTSAssumeRole) IsExpired() bool {
+	return s.stsAssumeRole.IsExpired()
+}
+
+// STSClient contains http.client configuration need it by STSAssumeRole
+var STSClient = PrepareSTSClient()
+
 func newMcsCredentials(accessKey, secretKey, location string) (*credentials.Credentials, error) {
-	return credentials.NewSTSAssumeRole(getMinIOServer(), credentials.STSAssumeRoleOptions{
+	stsEndpoint := getMinIOServer()
+	if stsEndpoint == "" {
```

Comment:
```suggestion
	if string.TrimSpace(stsEndpoint) == "" {
```
else `" "` will not enter here, same for the ones below

Reply:
how is this solved?